File: BE/src/main/java/com/be/service/FeedBackService.java
package com.be.service;

import com.be.dto.request.FeedBackRequest;
import com.be.dto.response.FeedBackResponse;
import com.be.entity.User;

import java.util.List;
import java.util.UUID;

public interface FeedBackService {
    FeedBackResponse createFeedBack(
            User currentUserId,
            UUID registrationId,
            FeedBackRequest request
    );

    List<FeedBackResponse> getMyFeedBacks(UUID currentUserId);

    List<FeedBackResponse> getFeedBacksByBuddy(
            User currentUser,
            UUID buddyId
    );
}

File: FE/Phimium-FE/src/services/feedbackService.js
import http from '@/services/http.js'

const feedbackService = {
  createFeedback: (registrationId, payload) => http.post(`/feedback/registrations/${registrationId}`, payload),

  getMyFeedbacks: () => http.get('/feedback/me'),

  getFeedbacksByBuddy: (buddyId) => http.get(`/feedback/buddies/${buddyId}`),
}

export default feedbackService
